### PR TITLE
Add new domain handling commands, fix existing ones, other bugfixes.

### PR DESCRIPTION
**Pull request by TuxSH**
Also adds domain support to sm.h.

### CODE REVIEW

File: nx/include/switch/ipc.h [hunk 1]
```diff
@@ -377,11 +379,11 @@ static inline Result ipcQueryPointerBufferSize(Handle session, size_t *size) {
         IpcParsedCommand r;
         ipcParse(&r);
 
-        struct ipcQueryPointerBufferSizeResponse {
+        struct {
             u64 magic;
             u64 result;
             u32 size;
-        } *raw = (struct ipcQueryPointerBufferSizeResponse*)r.Raw;
```
Comment (fincs) [inline nx/include/switch/ipc.h]: This breaks C++

File: nx/include/switch/ipc.h [hunk 2]
```diff
@@ -420,11 +422,11 @@ static inline Result ipcConvertSessionToDomain(Handle session, u32* object_id_ou
         IpcParsedCommand r;
         ipcParse(&r);
 
-        struct ipcConvertSessionToDomainResponse {
+        struct {
             u64 magic;
             u64 result;
             u32 object_id;
-        } *raw = (struct ipcConvertSessionToDomainResponse*) r.Raw;
```
Comment (fincs) [inline nx/include/switch/ipc.h]: Same thing here

File: nx/include/switch/services/sm.h [hunk 3]
```diff
@@ -68,9 +133,14 @@ static inline void serviceClose(Service* s) {
     switch (s->type) {
 
     case ServiceType_Normal:
+    case ServiceType_Domain:
```
Comment (fincs) [inline nx/include/switch/services/sm.h]: What happens if you create a domain from an overriden service? It shouldn't use svcCloseHandle here.

Comment (TuxSH) [reply nx/include/switch/services/sm.h]: You shouldn't convert an overriden service to a domain to begin with...

Do overriden domains exist/are they desirable?

Comment (fincs) [reply nx/include/switch/services/sm.h]: FS comes to mind as a service that might be overriden; and that may benefit from domains iirc.

Comment (plutooo) [reply nx/include/switch/services/sm.h]: Yeah we need to be able to override domains (but not subdomains)

**Pull request merged**